modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java

```diff
@@ -293,6 +293,52 @@ public String toEnumValue(String value, String datatype) {
         }
     }
 
+    /**
+     * Update datatypeWithEnum for array container
+     *
+     * @param property Codegen property
+     */
+    protected void updateDataTypeWithEnumForArray(CodegenProperty property) {
+        CodegenProperty baseItem = property.items;
+        while (baseItem != null && (Boolean.TRUE.equals(baseItem.isMap)
+                || Boolean.TRUE.equals(baseItem.isArray))) {
+            baseItem = baseItem.items;
+        }
+        if (baseItem != null) {
+            // set both datatype and datetypeWithEnum as only the inner type is enum
+            System.out.println("Marker: 1");
+            System.out.println("Replace: " + property.datatypeWithEnum + " from " + baseItem.baseType + " to " + toEnumName(baseItem));
+            System.out.println("Property info: " + property.isArray + " // " + property.isContainer + " // " + property.isFreeFormObject + " // " + property.isMap);
+            System.out.println("Property: "+ property.dataType);
+            System.out.println("Base ITem: " + baseItem.isMap + " // " + baseItem.baseName + " // " + baseItem.containerType + " // " + baseItem.dataType);
+            System.out.println("Base Type: "+ baseItem.jsonSchema);
+            System.out.println("Property: "+ property.jsonSchema);
+
+            /* 
+             * Note: There are cases where we have datatypeWithEnum == Array<{ [key: string]: string}
+             * In these cases, we then have Array <{ [key: EnumName]: EnumName}> - which is invalid typescript
+             * To protect agains this we first replace [key: string] with a special/reserved placeholder (i.e. *[key]* )
+             */
+            property.datatypeWithEnum = property.datatypeWithEnum.replace("[key: string]", "*key*")
```

Comment:
```
*key*
```
can be constant with PLACEHOLDER name.

#PLACEHOLDER# or something else.

Reply:
Actually using regular expression with replaceAll might help avoiding placeholder. We always looking to replace last occurence. E.g: `EnumName}>$`

Reply:
Is it always the last occurrence?

Wondering whether there are more complicated examples where this does not hold - mainly thinking about cases where the same type appears twice (e.g., `Tuple<string, string>` to `Tuple<FancyEnum, FancyEnum>`).

In cases like this, our model is anyway broken because we lose type information when generating the initial type.